src/Service/CoverStore/CloudinaryCoverStoreService.php

```diff
@@ -25,6 +25,20 @@
  */
 class CloudinaryCoverStoreService implements CoverStoreInterface
 {
+    /**
+     * Use to make paginated search (off-set in cloudinary).
+     *
+     * This is an array to ensure the next cursor used is for the requested search. So they are indexed by search query.
+     *
+     * @var array
+     */
+    private array $nextCursor = [];
```

Comment:
Having this on the class level is not a good idea. It opens the service to unexpected behavior and potential memory leaks in long running processes because this array is never reset.